### ISSUE TITLE
Altair charts not produced when executing notebook from command line and saving as html

### ISSUE THREAD

**Issue by Alcampopiano** [altair-viz/altair#1040]
Hi Jake and all,

I'm batch executing a bunch of notebooks from the command line with nbconvert. Once executed they are converted to HTML files. For some reason, instead of rendered charts,  I get the "vega warning" discussed in the Altair documentation. The executed notebook looks fine as all cells have run and the charts are there. If I export to HTML from the GUI, things are fine as well.

I'm using the following system call:
 `/home/allan/anaconda3/bin/jupyter nbconvert --to html my_notebook.ipynb`

The warning is the usual:
```
<VegaLite 2 object>

If you see this message, it means the renderer has not been properly enabled
for the frontend that you are using. For more information, see
https://altair-viz.github.io/user_guide/troubleshooting.html
```

I have tried:
`alt.renderers.enable('jupyterlab');` and `alt.renderers.enable('default');`

**Comment by Alcampopiano** [comment]
Is this the same issue? I'm not looking to retain interactivity in the html file. Rather, just looking to be able to convert to html with nbconvert without losing the Altair charts.

**Comment by jakevdp** [comment]
Yes, I think of them as the same issue: they're both rooted in the fact that nbconvert does not know how to render Altair plots.

**Comment by Alcampopiano** [comment]
Ah okay. Cool. I have like 60 notebooks to convert so the command line would be helpful of course. I will try to think of some other way to do mass conversion until using nbconvert is possible.

As usual, thanks Jake.

Al

**Comment by Alcampopiano** [comment]
The odd thing is that if I use the GUI's "restart kernel and run all cells" *and then* use nbconvert at the commandline, the html looks fine, charts and all.

**Comment by pybokeh** [comment]
@Alcampopiano I can convert my jupyter notebooks to html with Altair charts embedded fine.  The issue is, you lose interactivity.  For example, I have enabled tooltips, but they don't work when embedded in html file.  Perhaps Jake misunderstood you or didn't realize Altair charts will render in html, but they will be stripped of interactivity until nbconvert is updated for Altair.  That is my understanding anyways.  If yours did not render, it may have been a network glitch where the required javascript libraries didn't get loaded.  At work, my company's firewall blocked the URLs that host the Altair javascript libraries and I had to ask our security department to unblock them.  Then my Altair charts rendered fine.

**Comment by Alcampopiano** [comment]
@pybokeh Thanks for giving some hope. I will test things out from home tonight to see if the result is different. Appreciate your comments.

**Comment by Alcampopiano** [comment]
@pybokeh I have discovered that it's only when I use `execute` as part of the nbconvert command that the Altair charts fail to appear. If I use nbconvert to convert to HTML, but I leave out the `execute` flag, the Altair charts show up. Of course this doesn't help me since I would have to execute dozens and dozens of notebooks manually. What do you think of this?

<img src="https://media.giphy.com/media/WS0UKOeMzBXKaoCQfr/giphy.gif" width="800" />

### works (but depends on having run all cells from the GUI)
`jupyter nbconvert --to html my_notebook.ipynb
`
### does not work
`jupyter nbconvert --execute --to html my_notebook.ipynb`

**Comment by pybokeh** [comment]
@Alcampopiano I understand now.  Sorry I don't know enough about the internals of nbconvert.  I would have to look at the open issues on nbconvert compatibility with Altair.  Off the top of my head, I don't know of a workaround to execute all the cells of all the notebooks you need to convert.

**Comment by Alcampopiano** [comment]
@pybokeh No problem. I appreciate the comment. Just to be clear, you are not having this problem though right? Just trying to clarify if this is a known issue as Jake suggested, or something anomalous on my end.

**Comment by pybokeh** [comment]
@Alcampopiano Sorry forgot to mention, I am having the same problem as you describe with the --execute flag.  So it is not only you.

**Comment by Alcampopiano** [comment]
Thanks very much. Good to know.

**Comment by Alcampopiano** [comment]
@pybokeh Check out this great workaround https://github.com/altair-viz/altair/issues/329 . Now we can use `execute` from the command line *and* retain interactivity in the HTML versions of the notebook.

**Comment by jakevdp** [comment]
Some frontends add a static PNG to the cell outputs at render-time, so that HTML export will show a static plot (e.g. the notebook renderer here https://github.com/vega/ipyvega/blob/master/src/index.ts#L65-L73) This is not the case with all frontends, though.

**Comment by Alcampopiano** [comment]
Thanks Jake. That makes sense. Until nbconvert is updated, the workaround in https://github.com/altair-viz/altair/issues/329 solves my issue.
